### DISCUSSION TITLE
Whisperflo VST?

### DISCUSSION THREAD

**Discussion by jreising326** [tagyoureit/nodejs-poolController#778] [RS485]
Newbie with new Pump and new njsPC install.  Using the RS485 Dongle and Pentair Cable, for now just trying to get the Pump to turn on.  I have it configured in Dashpanel as Intelliflo VS and am getting setPumpRPMAsync errors.  Just focusing on getting the pump to run and make sure it's wired up and configured correctly, and to make sure this pump is actually supported.  I have 3 speeds setup as Features, I get no errors when I set the speed to zero/turn off the pump, so I think it is communicating OK.  Pump says it's in Auto Mode and is setup for "Direct Control".  Any and all advice is appreciated!  I saw a few posts discussion the differences but didn't see anything confirming someone has this working with the Whisperflow VST.  Any advice or info is appreciated.  Thanks, Joe

**Comment by rstrouse** [comment]
The rumor is that this pump communicates using the same protocol as the IntelliFlo VS.  I also believe there are others that have this pump working with njsPC.  Are you certain you have addressed the pump correctly.  You should verify the yellow/green wires are connected to RS485 A/B correctly.

**Reply by jreising326** [reply]
Thank you for confirming!  As you were replying I found in a post online that the "External Control" Function is for Digital Inputs and not RS485.  So I turned off External Control at the Pump, Cycled Powered, then hit the "Start" button and now njsPC is controlling the Pump.  The console is still reporting the same error message and I assume it's just trapping something and not a problem.  So yes I can also confirm this Pump Works once you understand what "mode" to put it in, and use the proper sequence of events.  Those events, at least in my case is 1)  njsPC running with Pump Setting to OFF in njsPC and RS485 connected.  2)  Cycle power on Pump, hit Start, then turn pump on from njsPC.  Intelliflo VS selected as the Pump in njsPC and Address Set to 1 in my case.  Over documenting because I spent most of the weekend chasing my tail..  hope this helps someone else!

**Comment by evercape** [comment]
I have the same pump, use the Pentair cable and an USB dongle. I can't even get a serial tool such as SerialTools or SerialMonster show the incoming data from the pump.

What settings do you use on the Pump? 

Thanks for your help.


![Document1](https://github.com/tagyoureit/nodejs-poolController/assets/4627761/0f5d55c5-3429-4f48-b24e-67da83584cc1)
![serialtools](https://github.com/tagyoureit/nodejs-poolController/assets/4627761/28de8c6a-1621-4a73-bf19-c460b5202215)
![SerialMonster](https://github.com/tagyoureit/nodejs-poolController/assets/4627761/8cfd1769-2ef2-4853-bc8e-7427467140e4)

**Reply by evercape** [reply]
Just to be clear the Pentair comms cable is connected directly to the pump which is on pump address=1

**Reply by jreising326** [reply]
You have to have all the right settings and then follow the sequence above.  I don't think Serial Monster or any Serial Port Sniffer will be that useful, pretty sure I tried something similar.  I think the Pump is waiting for a specific Command, once all settings are correct, and you do the Power Cycle.  Do not enable External Control, use Intelliflo VS in njPC, make sure wires aren't backwards, cycle power on pump with njPC "talking", and it should work.  Once njPC is in control, the touchpad is disabled other than Start/Stop I believe.

**Reply by rstrouse** [reply]
You are correct.   Please cycle the start/stop button after you have disconnected your external control.  Once the RS485 takes over all other forms of control are disabled.
